jetbrains-core/src/software/aws/toolkits/jetbrains/services/sqs/toolwindow/SendMessagePane.kt

```diff
@@ -2,8 +2,119 @@
+    var statusLabel = JLabel()
+    private var isFifo = true
+    private val cache = MessageCache.getInstance()
+
+    init {
+        setUp()
+        sendButton.addActionListener {
+            launch { sendMessage() }
+        }
+        clearButton.addActionListener {
+            inputText.text = ""
+            deduplicationId.text = ""
+            groupId.text = ""
+            cache.setMessage(queue.queueUrl, inputText.text)
+            confirmationPanel.isVisible = false
+        }
+    }
+
+    private fun setUp() {
+        if (!queue.queueName.endsWith(".fifo")) {
+            fifoComponent.isVisible = false
+            isFifo = false
+        }
+        emptyDeduplicationLabel.isVisible = false
+        emptyGroupLabel.isVisible = false
+        emptyBodyLabel.isVisible = false
+        confirmationPanel.apply {
+            isVisible = false
+            setContent(statusLabel)
+        }
+
+        deduplicationContextHelp.icon = AllIcons.General.ContextHelp
+        HelpTooltip().apply {
+            setDescription(message("sqs.message.deduplication_id.tooltip"))
+            installOn(deduplicationContextHelp)
+        }
+        groupContextHelp.icon = AllIcons.General.ContextHelp
+        HelpTooltip().apply {
+            setDescription(message("sqs.message.group_id.tooltip"))
+            installOn(groupContextHelp)
+        }
+
+        inputText.emptyText.text = message("sqs.send.message.body.empty.text")
+        deduplicationId.emptyText.text = message("sqs.required.empty.text")
+        groupId.emptyText.text = message("sqs.required.empty.text")
+
+        inputText.text = cache.getMessage(queue.queueUrl)
+    }
+
+    suspend fun sendMessage() {
+        if (validateFields()) {
+            try {
+                withContext(Dispatchers.IO) {
+                    val messageId = client.sendMessage {
+                        it.queueUrl(queue.queueUrl)
+                        it.messageBody(inputText.text)
+                        if (isFifo) {
+                            it.messageDeduplicationId(deduplicationId.text)
+                            it.messageGroupId(groupId.text)
+                        }
+                    }.messageId()
+
+                    statusLabel.text = message("sqs.send.message.success", messageId)
+                }
+                cache.setMessage(queue.queueUrl, inputText.text)
+            } catch (e: Exception) {
+                statusLabel.text = message("sqs.failed_to_send_message")
+            }
+            confirmationPanel.isVisible = true
+        }
+    }
+
+    private fun validateFields(): Boolean {
+        emptyBodyLabel.isVisible = inputText.text.isEmpty()
+        if (isFifo) {
+            emptyDeduplicationLabel.isVisible = deduplicationId.text.isEmpty()
+            emptyGroupLabel.isVisible = groupId.text.isEmpty()
+        }
+
+        return !(emptyBodyLabel.isVisible || emptyDeduplicationLabel.isVisible || emptyGroupLabel.isVisible)
```

Comment:
This check should be based on what isVisible is based on, not the state of isVisible. You can save the isEmpty checks to booleans then reuse them